### PR TITLE
[fix] making start and stop async

### PR DESCRIPTION
**Pull request by hengchu**
Hopefully this will help with test failure on windows.

### CODE REVIEW

File: src/monitor.ts [hunk 1]
```diff
@@ -160,27 +160,37 @@ export class Monitor {
...
    * Start the embedded HTTP server to process requests on the given port.
    */
-  public start(port: number) {
+  public async start(port: number): Promise<void> {
     this.timer = this.createPingTimer();
     this.ping();
 
-    this.server = http.createServer(this.app);
-    this.server.listen(port);
+    const promise = new Promise<void>(resolve => {
```
Comment (danqing) [inline src/monitor.ts]: just return here? (same below)

File: src/tests/monitor.test.ts [hunk 2]
```diff
@@ -199,3 +193,24 @@ test.serial(
...
   t.is(triedToReviveServer1, false);
 });
+
+test.serial('monitor stop should be idempotent', async t => {
+  const context = t.context as ITestContext;
+
+  await context.monitor.stop();
+  await context.monitor.stop();
+
+  t.pass();
```
Comment (danqing) [inline src/tests/monitor.test.ts]: hmm what's this testing here?

Comment (hengchu) [reply src/tests/monitor.test.ts]: Just testing double stop doesn't throw...

File: src/monitor.ts (hunk 1 above)
Comment (hengchu) [reply src/monitor.ts]: I want the promise to only resolve after `listen()` has finished, which occurs in the callback, would returning here have that behavior?

Comment (danqing) [reply src/monitor.ts]: afaik yes. assigning it to a variable shouldn't change anything.

File: src/tests/monitor.test.ts (hunk 2 above)
Comment (danqing) [reply src/tests/monitor.test.ts]: lol okay

File: src/monitor.ts (hunk 1 above)
Comment (hengchu) [reply src/monitor.ts]: oh oh I misunderstood. I will fix this

**Pull request merged**